spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/ConverterAwareMappingSpannerEntityWriter.java

```diff
@@ -322,7 +326,12 @@ private static boolean attemptSetIterableValue(Iterable<Object> value,
 
 	@SuppressWarnings("unchecked")
 	private static <T> boolean attemptSetSingleItemValue(Object value, Class<?> sourceType,
-			ValueBinder<WriteBuilder> valueBinder, Class<T> targetType, SpannerCustomConverter writeConverter) {
+			ValueBinder<WriteBuilder> valueBinder, Class<T> targetType,
+			SpannerCustomConverter writeConverter, Type.Code annotatedColumnItemType) {
+		if (annotatedColumnItemType == Type.Code.JSON) {
```

Comment:
Sorry, I think my previous advice about moving the JSON check logic here was not quite correct. I didn't realize you don't have access to the column type code in here, and then it doesn't seem quite right to have to pass `null` each time for the other usages.

I think we should go back to doing the `valueBinder.to(..)` back in the `writeProperty(..)` method, if you feel that way is simpler. Sorry to go back and forth.